services/web/server/src/simcore_service_webserver/projects/projects_db.py

```diff
@@ -819,10 +656,73 @@ async def check_delete_project_permission(self, user_id: int, project_uuid: str)
                     conn, user_id, project_uuid, for_update=True
                 )
                 # if we have delete access we delete the project
-                user_groups: list[RowProxy] = await self.__load_user_groups(
+                user_groups: list[RowProxy] = await self._list_user_groups(
                     conn, user_id
                 )
-                _check_project_permissions(project, user_id, user_groups, "delete")
+                check_project_permissions(project, user_id, user_groups, "delete")
+
+    #
+    # Project TAGS
+    #
+
+    async def add_tag(self, user_id: int, project_uuid: str, tag_id: int) -> dict:
+        async with self.engine.acquire() as conn:
+            project = await self._get_project(conn, user_id, project_uuid)
+            # pylint: disable=no-value-for-parameter
+            query = study_tags.insert().values(study_id=project["id"], tag_id=tag_id)
+            user_email = await self._get_user_email(conn, user_id)
+            async with conn.execute(query) as result:
+                if result.rowcount == 1:
+                    project["tags"].append(tag_id)
+                    return convert_to_schema_names(project, user_email)
+                raise ProjectsException()
+
+    async def remove_tag(self, user_id: int, project_uuid: str, tag_id: int) -> dict:
+        async with self.engine.acquire() as conn:
+            project = await self._get_project(conn, user_id, project_uuid)
+            user_email = await self._get_user_email(conn, user_id)
+            # pylint: disable=no-value-for-parameter
+            query = study_tags.delete().where(
+                and_(
+                    study_tags.c.study_id == project["id"],
+                    study_tags.c.tag_id == tag_id,
+                )
+            )
+            async with conn.execute(query):
+                if tag_id in project["tags"]:
+                    project["tags"].remove(tag_id)
+                return convert_to_schema_names(project, user_email)
+
+    #
+    # Project HIDDEN column
+    #
+
+    async def set_hidden_flag(self, project_uuid: ProjectID, enabled: bool):
```

Comment:
Q: should this not re-use the replace or update method in the CRUD methods?

Reply:
That is an option but I think it would be better to have it as a separate sub-resource. Note ``ProjectDBApi`` implements a repository pattern that acts as interface on top of the database. We could add CRUD on subresources for convenience as well. 
In any case, the discussion is added to #3516